fdbrpc/FlowGrpcTests.cpp

```diff
@@ -162,6 +180,51 @@ TEST_CASE("/fdbrpc/grpc/stream_destroy") {
 // 	co_return;
 // }
 
+TEST_CASE("/fdbrpc/grpc/file_transfer") {
+	using platform::TmpFile;
+
+	// -- Server --
+	std::string server_address("127.0.0.1:50051");
+	FileTransferServiceImpl service;
+
+	grpc::ServerBuilder builder;
+	builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+	builder.RegisterService(&service);
+
+	std::unique_ptr<grpc::Server> server(builder.BuildAndStart());
+	std::cout << "Server listening on " << server_address << std::endl;
+
+	// -- Client --
+	auto channel = grpc::CreateChannel(server_address, grpc::InsecureChannelCredentials());
+	auto client = FileTransferClient(channel);
+	TmpFile src;
+	TmpFile dest;
+	std::cout << "Writing 1GB random bytes to source file.\n";
+	int bytes_written = 0;
+	for (int i = 0; i < 1024; ++i) {
+		const int size = i < 1023 ? 1024 * 1024 : (1024 * 1024) / 2; // Make last block smaller.
+		auto rand_str = generate_random_string(size);
+		src.append((const uint8_t*)rand_str.c_str(), size);
+		bytes_written += size;
+	}
+	std::cout << "Finished writing " << bytes_written << " bytes.\n";
+	std::cout << "Invoking gRPC File Transfer call.\n";
+	auto start = std::chrono::high_resolution_clock::now();
+	auto res = client.DownloadFile(src.getFileName(), dest.getFileName());
```

Comment:
Pass a checksum/CRC (https://www.evanjones.ca/tcp-and-ethernet-checksums-fail.html)? Fail the transfer if verification on receiving side doesn't match the source checksum/crc (like s3 does)?

Reply:
Yes. Totally. In my pending.